lib/mayaUsd/utils/utilComponentCreator.cpp

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright 2025 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <mayaUsd/utils/utilComponentCreator.h>
+
+#include <pxr/base/tf/diagnostic.h>
+#include <pxr/pxr.h>
+
+#include <maya/MGlobal.h>
+#include <maya/MString.h>
+
+#include <vector>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+namespace ComponentUtils {
+
+std::vector<std::string> getAdskUsdComponentLayersToSave(const std::string& proxyPath)
+{
+    // Ask via python what layers need to be saved for the component.
+    // With the maya api we can only return a string, so we concat the ids.
+    MString getLayersFromComponent;
+    getLayersFromComponent.format(
+        "def usd_component_creator_get_layers_to_save():\n"
+        "    import mayaUsd\n"
+        "    import mayaUsd.ufe\n"
+        "    from usd_component_creator_plugin import MayaComponentManager\n"
+        "    stage = mayaUsd.ufe.getStage(\"^1s\")\n"
+        "    if stage is None:\n"
+        "        return \"\"\n"
+        "    ids = MayaComponentManager.GetInstance().GetSaveInfo(stage)\n"
+        "    result = \"\"\n"
+        "    first = True\n"
+        "    for id in ids:\n"
+        "        if not first:\n"
+        "            result += \"|\"\n"
+        "        result += id\n"
+        "        first = False\n"
+        "    return result\n",
+        proxyPath.c_str());
+
+    if (MGlobal::executePythonCommand(getLayersFromComponent)) {
+        auto resultString = MGlobal::executePythonCommandStringResult(
+            "usd_component_creator_get_layers_to_save()");
+        MStringArray layerIds;
+        resultString.split('|', layerIds);
+        std::vector<std::string> toSave;
+        for (const auto& id : layerIds) {
+            toSave.push_back(id.asUTF8());
+        }
+        return toSave;
+    }
+    return {};
+}
+
+bool isAdskUsdComponent(const std::string& proxyShapePath)
+{
+    MString defineIsComponentCmd;
+    defineIsComponentCmd.format(
```

Comment:
Not for now, but I think we should eventually refactor all those Python functions into a "helper" sub-module of usd_component_creator and avoid redefining and reparsing these functions all the time. (Merged with Anthon's)